Bottom: 26f82a7df068debc0a294c65c2a161a8f0af560f
Top:    d9893c5a5f6b9b8aec32bd42957293b9edc6c86b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-06 19:46:08 +0000

Try not using no-holes dicer for drawing polygons




---

diff --git a/src/draw.c b/src/draw.c
index 1cacc5c..8983654 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2204,7 +2204,7 @@ DrawPour (LayerTypePtr Layer, PourTypePtr Pour, int unused)
   layernum = GetLayerNumber (PCB->Data, Layer);
   DrawPourLowLevel (Pour, NULL);
 #warning FIXME Later
-#if 0
+#if 1
   if (TEST_FLAG (CLEARPOLYFLAG, Pour))
     {
       r_search (PCB->Data->pin_tree, &Pour->BoundingBox, NULL,
@@ -2272,6 +2272,7 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
   else if (Polygon->Clipped)
     {
+#if 0
       if (!Polygon->NoHolesValid)
         {
           ComputeNoHoles (Polygon);
@@ -2285,6 +2286,15 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
             poly.Clipped = poly.Clipped->f;
           } while (poly.Clipped != Polygon->NoHoles);
         }
+#endif
+        {
+          PolygonType poly = *Polygon;
+          poly.Clipped = Polygon->Clipped;
+          do {
+            DrawPolygonLowLevel (&poly, NULL);
+            poly.Clipped = poly.Clipped->f;
+          } while (poly.Clipped != Polygon->Clipped);
+        }
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
       if (TEST_FLAG (FULLPOLYFLAG, Polygon))
